pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -238,38 +235,36 @@ contract IndexPool is BaseWeightedPool, ReentrancyGuard {
         _startGradualWeightChange(startTime, endTime, _getNormalizedWeights(), endWeights);
     }
 
-
-    function reweighTokens(address[] calldata tokens, uint256[] calldata desiredWeights) public{
+    function reweighTokens(address[] calldata tokens, uint256[] calldata desiredWeights) public authenticate {
         uint256 endTime = _poolState.decodeUint32(_END_TIME_OFFSET);
-        require(block.timestamp >= endTime ,"Weight change is already in process");
+        require(block.timestamp >= endTime, "Weight change is already in process");
         uint256 diff = 0;
         uint256 numTokens = tokens.length;
         InputHelpers.ensureInputLengthMatch(numTokens, desiredWeights.length);
 
         uint256 normalizedSum = 0;
         for (uint8 i = 0; i < numTokens; i++) {
-            if(desiredWeights[i] > _normalizedWeights[i]){
-                if(diff < desiredWeights[i].sub(_normalizedWeights[i])){
+            if (desiredWeights[i] > _normalizedWeights[i]) {
+                if (diff < desiredWeights[i].sub(_normalizedWeights[i])) {
                     diff = desiredWeights[i].sub(_normalizedWeights[i]);
                 }
-            }
-            else {
-                if(diff < _normalizedWeights[i].sub(desiredWeights[i])){
+            } else {
+                if (diff < _normalizedWeights[i].sub(desiredWeights[i])) {
                     diff = _normalizedWeights[i].sub(desiredWeights[i]);
                 }
             }
             normalizedSum = normalizedSum.add(desiredWeights[i]);
         }
         _require(normalizedSum == FixedPoint.ONE, Errors.NORMALIZED_WEIGHT_INVARIANT);
         uint256 change_time = ((diff.mulDown(_SECONDS_IN_A_DAY)).divDown(FixedPoint.ONE)) * 100;
-        updateWeightsGradually(block.timestamp, block.timestamp.add(change_time), desiredWeights);
+        _updateWeightsGradually(block.timestamp, block.timestamp.add(change_time), desiredWeights);
     }
 
     function reindexTokens(
         address[] calldata tokens,
         uint256[] calldata desiredWeights,
         uint256[] calldata minimumBalances
-    ) external {
+    ) external view authenticate {
```

Comment:
It's a view rn, but it will be not a view function in the future...

Reply:
Removed the `view` keyword as well as the `pure` keyword that got merged from main. The modifier `authenticate` does not work with a `pure` function